### PR TITLE
Allow bot and others to close stale issues

### PR DESCRIPTION
**Pull request by fejta**
/assign @BenTheElder @cjwagner 

If an issue has a lifecycle/rotten or lifecycle/stale label, allow fejta-bot to close it

ref https://github.com/kubernetes/test-infra/issues/5507

### CODE REVIEW

File: prow/plugins/lifecycle/close.go [hunk 1]
```diff
@@ -78,20 +92,31 @@ func handleClose(gc closeClient, log *logrus.Entry, e *github.GenericCommentEven
...
-			} else if merr != nil {
-				log.WithError(merr).Errorf("Failed IsMember(%s, %s)", org, commentAuthor)
-			} else {
-				log.WithError(err).Errorf("Failed AssignIssue(%s, %s, %d, %s)", org, repo, number, commentAuthor)
+	isAuthor := e.IssueAuthor.Login == commentAuthor
+
+	if !isAssignee && !isAuthor {
+		active, err := isActive(gc, org, repo, number)
+		if err != nil {
+			log.Infof("Cannot determine if issue is active: %v", err)
```
Comment (cjwagner) [inline prow/plugins/lifecycle/close.go]: The error already includes the context "could not determine if issue is inactive".

File: prow/plugins/lifecycle/close.go [hunk 2]
```diff
@@ -78,20 +92,31 @@ func handleClose(gc closeClient, log *logrus.Entry, e *github.GenericCommentEven
...
+			active = true // Fail active
+		}
+
+		if active {
+			// Try to assign the issue to the comment author
+			log.Infof("Assign to %s", commentAuthor)
+			if err := gc.AssignIssue(org, repo, number, []string{commentAuthor}); err != nil {
+				msg := "Assigning you to the issue failed."
+				if ok, merr := gc.IsMember(org, commentAuthor); merr == nil && !ok {
+					msg = "Only kubernetes org members may be assigned issues."
```
Comment (cjwagner) [inline prow/plugins/lifecycle/close.go]: This message shouldn't be Kubernetes specific.

File: prow/plugins/lifecycle/close_test.go [hunk 3]
```diff
@@ -61,6 +62,17 @@ func (c *fakeClientClose) AssignIssue(owner, repo string, number int, assignees
 	return nil
 }
 
+func (c *fakeClientClose) GetIssueLabels(owner, repo string, number int) ([]github.Label, error) {
+	if repo == "error" {
+		return nil, errors.New("issue label 500")
```
Comment (cjwagner) [inline prow/plugins/lifecycle/close_test.go]: This error case is unused.

File: prow/plugins/lifecycle/close_test.go [hunk 4]
```diff
@@ -157,9 +170,31 @@ func TestCloseComment(t *testing.T) {
...
+		{
+			name:          "close by other person, stale issue",
+			action:        github.GenericCommentActionCreated,
+			state:         "open",
+			body:          "/close",
+			commenter:     "non-member",
+			labels:        []string{"lifecycle/stale"},
+			shouldClose:   true,
+			shouldComment: false,
+			shouldAssign:  false,
```
Comment (cjwagner) [inline prow/plugins/lifecycle/close_test.go]: `shouldAssign` is not used.

File: prow/plugins/lifecycle/close_test.go (hunk 3 above)
Comment (fejta) [reply prow/plugins/lifecycle/close_test.go]: Done

File: prow/plugins/lifecycle/close.go (hunk 2 above)
Comment (fejta) [reply prow/plugins/lifecycle/close.go]: Done

File: prow/plugins/lifecycle/close.go (hunk 1 above)
Comment (fejta) [reply prow/plugins/lifecycle/close.go]: Done

File: prow/plugins/lifecycle/close_test.go (hunk 4 above)
Comment (fejta) [reply prow/plugins/lifecycle/close_test.go]: Not sure what you mean here... It is definitely used later in the function.

Comment (cjwagner) [reply prow/plugins/lifecycle/close_test.go]: You're right, I misread the diff.

**Pull request merged**